File: Python/estoque/src/controller/estoque_controller.py
from fastapi import APIRouter
from service import estoque_service
from dto.estoque_DTO import EstoqueDTO;
from decimal import Decimal

router = APIRouter()

@router.get("/estoqueBaixo")
def find_all_estoque_baixo():
	return estoque_service.find_all_estoque_baixo()

@router.put("/{produto_id}/{quantidade}")
def atualiza_estoque(produto_id: int, quantidade: Decimal):
	return estoque_service.atualiza_estoque(produto_id, quantidade)

@router.get("/{id}")
def get_by_id(id: int):
	return estoque_service.get_by_id(id)

@router.get("")
def get(page: int = 0, size: int = 10, sort: str = "id,ASC"):
	return estoque_service.get(page, size, sort)

@router.put("/{id}")
def update(id: int, estoqueDTO: EstoqueDTO):
	return estoque_service.update(id, estoqueDTO)

@router.post("")
def save(estoqueDTO: EstoqueDTO):
	return estoque_service.save(estoqueDTO)

File: Java/estoque/src/main/java/br/tcc/estoque/controller/EstoqueController.java
package br.tcc.estoque.controller;

import java.math.BigDecimal;

import org.springframework.data.domain.Pageable;
import org.springframework.data.web.PageableDefault;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import br.tcc.estoque.record.EstoqueRecord;
import br.tcc.estoque.service.EstoqueService;
import jakarta.validation.Valid;

@RestController
@RequestMapping("/estoque")
public class EstoqueController {
	
	public final EstoqueService estoqueService;
	
	public EstoqueController(EstoqueService estoqueService) {
		this.estoqueService = estoqueService;
	}
	
	@GetMapping("/estoqueBaixo")
	public ResponseEntity<Object> findAllEstoqueBaixo() {
		return estoqueService.findAllEstoqueBaixo();
	}

	@PutMapping("/{idProduto}/{quantidade}")
	public ResponseEntity<Object> atualizaEstoque(@PathVariable final Long idProduto, @PathVariable final BigDecimal quantidade) {
		return estoqueService.atualizaEstoque(idProduto, quantidade);
	}
	
	@GetMapping("{id}")
	public ResponseEntity<Object> getById(@PathVariable final Long id) {
		return estoqueService.getById(id);
	}
	
	@GetMapping
	public ResponseEntity<Object> get(@PageableDefault(page = 0, size = 10) final Pageable page) {
		return estoqueService.get(page);
	}
	
	@PutMapping("{id}")
	public ResponseEntity<Object> update(@PathVariable final Long id, @RequestBody @Valid final EstoqueRecord dto) {
		return estoqueService.update(id, dto);
	}
	
	@PostMapping
	public ResponseEntity<Object> save(@RequestBody @Valid final EstoqueRecord dto) {
		return estoqueService.save(dto);
	}
	
}